### DISCUSSION TITLE
Analytical Target Cascading in openMDAO ?

### DISCUSSION THREAD

**Discussion by SebastianEEE** [OpenMDAO/OpenMDAO#3108] [Q&A]
Dear openMDAO-team,

let me start off by congratulating you for your awesome work. I have been fascinated by your steady progress and great capabilities. I would really like to use openMDAOfor my PhD project in which I try to optimize a vehicle efficiently. 

I came across various MDO architectures and specifically multi-level / hierarchical architectures such as ATC or ALC seem to be a good fit for my requirements. However, it seems to me, that other researchers / practitioners have only used monolithic or bi-level architectures in openMDAO sofar. 

Is there an inherent reason / obstacle, why openMDAO may not fit multilevel architectures (such as ATC) ? From your experience, is it advisable to try to implement ATC in openMDAO ?

Thank you very much for your answer.

**Comment by JustinSGray** [comment]
Way back in the days of OpenMDAO V0, we did a lot of work on hierarchical architectures. But that work fell to the wayside a bit when we saw that we were getting much better performance from monolithic problems with full analytic derivatives. Since that time, a number of MDO community members, particularly in Europe have started looking into hierarchical architectures again. 

We don't officially support them in OpenMDAO, but we don't not support them either. Back in 2020, we did some [basic work on sub-problems ](https://github.com/OpenMDAO/RevHack2020/blob/master/solution_approaches/sub_problems.md). Since then, in part due to revived interest in hierarchical architectures by the community, the dev team build in official support for SubModels via the the [SubModelComp](https://openmdao.org/newdocs/versions/3.30.0/features/building_blocks/components/submodel_comp.html?highlight=subproblem). 

Its been a while since I looked at ATC, so I don't remember the specifics of how its set up any more. But you could probably use a series of  sub-problems to build the hierarchy you wanted.

**Reply by robfalck** [reply]
Our current implementation of subproblems is such that analytic derivatives _through_ a sub-optimization are not available to the parent driver.  If this is a requirement of the way your ATC architecture is setup, then you're going to be limited to using finite difference. It will be subject to the same limitations that apply to finite-difference around nonlinear solvers.

We are working on expanding our support of multi-level drivers since there's so much demand for them, but for now we're going to be limited to this "gradient-free over gradient-based" approach. This will be fine when the top level driver is something like a Genetic algorithm, a simplex method, or a design space exploration/DOE.
